src/schema/posts.ts

```diff
@@ -833,6 +854,50 @@ export const resolvers: IResolvers<any, Context> = {
       });
       return { _: true };
     },
+    submitExternalLink: async (
+      _,
+      {
+        sourceId,
+        url,
+        commentary,
+      }: { sourceId: string; url: string; commentary: string },
+      ctx,
+    ): Promise<GQLEmptyResponse> => {
+      await ctx.con.transaction(async (manager) => {
+        await ensureSourcePermissions(ctx, sourceId, SourcePermissions.Post);
+        const cleanUrl = standardizeURL(url);
+        if (!isValidHttpUrl(cleanUrl)) {
+          throw new ValidationError('URL is not valid');
+        }
+
+        const existingPost = await manager.getRepository(ArticlePost).findOne({
+          select: ['id'],
+          where: [
+            { url: cleanUrl, deleted: false },
+            { canonicalUrl: cleanUrl, deleted: false },
```

Comment:
It will simply throw an error which we'll have to catch FE wise.
If a post is deleted for whatever reason we shouldn't allow it back (even as private) 

I'll double check with product as well.

Reply:
Not sure where the error is getting thrown here. Should we catch it here in BE too? cc @idoshamun

Reply:
Added the actual throw and test case for deleted posts.